### PR TITLE
API definitions generator

### PR DESCRIPTION
**Pull request by mmbabaev**

### CODE REVIEW

**Comment by fomkin** [comment]
Can you make pr into `full-api-coverage`?

File: build.sbt [hunk 1]
```diff
@@ -17,6 +17,8 @@ val commonSettings = Seq(
   }
 )
 
+val lift_json = "net.liftweb" %% "lift-json" % "2.3"
```
Comment (fomkin) [inline build.sbt]: Where it used?

File: project/ApiDefinitions.scala [hunk 2]
```diff
@@ -9,30 +9,23 @@ object ApiDefinitions {
   }
 
   val modules = Seq(
```
Comment (fomkin) [inline project/ApiDefinitions.scala]: I think this object should be dropped.

File: project/ApiDefinitionsGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,164 @@
+
+import scala.collection.mutable.ArrayBuffer
+import scala.util.parsing.json.JSON
```
Comment (fomkin) [inline project/ApiDefinitionsGenerator.scala]: This API is deprecated in Scala 2.11.

File: rethinkdb_data/log_file [hunk 4]
```diff
@@ -0,0 +1,16 @@
+2015-11-18T20:33:06.981222000 0.502685s notice: Recursively removing directory /Users/Mihail/Desktop/githubRepos/my/scala-reql/rethinkdb_data/tmp
```
Comment (fomkin) [inline rethinkdb_data/log_file:1]: This file should be added to `.gitignore`

File: scala-reql-akka/src/main/scala/reql/akka/ReqlActor.scala [hunk 5]
```diff
@@ -73,6 +72,8 @@ trait ReqlActor[Data] extends Actor with ReqlContext[Data] {
 
     var closed = false
 
+
```
Comment (fomkin) [inline scala-reql-akka/src/main/scala/reql/akka/ReqlActor.scala]: Do not make useless changes.

File: term_info.json [hunk 6]
```diff
@@ -0,0 +1,1391 @@
+{
```
Comment (fomkin) [inline term_info.json:1]: I think this file should be added as part of submodule.

File: project/ApiDefinitionsGenerator.scala [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+import scala.io.Source
+object ApiDefinitionsGenerator {
+
+  def modules(): Seq[module] = {
+    //todo: read term_info.json from java driver
+    val jsonFilename = "term_info.json"
+    var modules = ArrayBuffer[module]()
```
Comment (fomkin) [inline project/ApiDefinitionsGenerator.scala]: 1. Don't use `var` for mutable collection definition.
2. Don't use mutable collections for parsing cases :)

File: project/ApiDefinitionsGenerator.scala [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    }
+  }
+
+  def genFunctions(signaturesAny: Any,
+                   optArgs: ArrayBuffer[ArgOrOpt],
+                   includeInTypes: ArrayBuffer[Top]): ArrayBuffer[fun] = {
+
+    signaturesAny match {
+      case Some(signatures: List[List[Any]]) =>
+        var functions = ArrayBuffer[fun]()
```
Comment (fomkin) [inline project/ApiDefinitionsGenerator.scala]: See comment above.

File: project/ApiDefinitionsGenerator.scala [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+        for (signature <- signatures) {
+          //todo: implement signatures with function type
+          if (!signature.contains("T_FUNC0") && !signature.contains("T_FUNC1") && !signature.contains("T_FUNC2") && !signature.contains("T_FUNCX")) {
+            if (signature.isEmpty) {
+              functions += fun()
```
Comment (fomkin) [inline project/ApiDefinitionsGenerator.scala]: Huh?

File: project/ApiDefinitionsGenerator.scala [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+import scala.io.Source
+object ApiDefinitionsGenerator {
+
+  def modules(): Seq[module] = {
+    //todo: read term_info.json from java driver
+    val jsonFilename = "term_info.json"
+    var modules = ArrayBuffer[module]()
+
+    var jsonString = ""
+    for (line <- Source.fromFile(jsonFilename).getLines()) {
```
Comment (fomkin) [inline project/ApiDefinitionsGenerator.scala]: Why you use `getLines` api?

File: project/ApiDefinitionsGenerator.scala (hunk 7 above)
Comment (mmbabaev) [inline project/ApiDefinitionsGenerator.scala]: What is the right way to parse couple of objects? I created mutable collection for adding modules one by one :)

File: project/ApiDefinitionsGenerator.scala (hunk 9 above)
Comment (mmbabaev) [inline project/ApiDefinitionsGenerator.scala]: smth wrong? Understandable code?)

File: project/ApiDefinitionsGenerator.scala (hunk 7 above)
Comment (fomkin) [inline project/ApiDefinitionsGenerator.scala]: It is a bad way. Scala forces immutability and it fits to transformation cases.

``` scala
val modules = JSON.parseFull(jsonString) match {
  case Some(map: Map[String, Any]) =>
    map.keys collect {
      case key @ ("EQ" | "NE" | "LT" | "GT" | "BRANCH" | "OR" | "AND" | "ADD" | "SUB" | "MATCH" | "OBJECT") =>
        genModule(map.get(key), key)
      }
    }
  case _ => 
    Seq.empty[Module]
}
```

File: project/ApiDefinitionsGenerator.scala (hunk 9 above)
Comment (fomkin) [inline project/ApiDefinitionsGenerator.scala]: I don't understand behavior you expect.

**Pull request closed without merging**